src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -852,6 +840,26 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
         return c;
     }
 
+    public static StandardCredentials resolveCredentials(Job job, String credentialsId, String url) {
```

Comment:
I suggest to reduce the scope to `private`. Otherwise this method gets part of the public API and has to be maintained like this. Or was this intended?